### PR TITLE
[LLM] whisper model transformer int4 verification and example

### PR DESCRIPTION
**Pull request by leonardozcm**
## Description

Which is the autoclass of whisper.

```python
from transformers import WhisperProcessor, AutoModelForSpeechSeq2Seq
from bigdl.llm.transformers import AutoModelForSpeechSeq2Seq as BigdlAutoModelForSpeechSeq2Seq
from datasets import load_dataset

model_path = './whisper-tiny.en'

# load model and processor
processor = WhisperProcessor.from_pretrained(model_path)
# model = AutoModelForSpeechSeq2Seq.from_pretrained(model_path)


# load dummy dataset and read audio files
ds = load_dataset("hf-internal-testing/librispeech_asr_dummy", "clean", split="validation")
sample = ds[0]["audio"]
input_features = processor(sample["array"], sampling_rate=sample["sampling_rate"], return_tensors="pt").input_features 

# generate token ids
model = AutoModelForSpeechSeq2Seq.from_pretrained(model_path)
model.config.forced_decoder_ids = None
import time
st = time.time()
predicted_ids = model.generate(input_features)
time_0 = time.time() - st
# decode token ids to text

transcription = processor.batch_decode(predicted_ids, skip_special_tokens=True)

print(transcription)

# generate token ids
model = BigdlAutoModelForSpeechSeq2Seq.from_pretrained(model_path, load_in_4bit=True)
model.config.forced_decoder_ids = None
import time
st = time.time()
predicted_ids = model.generate(input_features)
time_1 = time.time() - st
# decode token ids to text

transcription = processor.batch_decode(predicted_ids, skip_special_tokens=True)

print(transcription)

print(f"fp32 time {time_0}s, int4 time {time_1}")
```
output:
```log
[' Mr. Quilter is the apostle of the middle classes, and we are glad to welcome his gospel.']
[" Mr. Quilter is the apostle of the middle classes and we're glad to welcome his Gospel."]
fp32 time 0.8264369964599609s, int4 time 0.20679783821105957
```

### CODE REVIEW

**Comment by jason-dai** [comment]
Have you tested it on client?

**Comment by leonardozcm** [comment]
> Have you tested it on client?

Not yet, will post client result later.

File: python/llm/example/langchain/native_int4/voiceassistant.py [hunk 1]
```diff
@@ -47,6 +48,7 @@ def prepare_chain(args):
 
     # We use our BigdlNativeLLM to subsititute OpenAI web-required API
     callback_manager = CallbackManager([StreamingStdOutCallbackHandler()])
+
```
Comment (shane-huang) [inline python/llm/example/langchain/native_int4/voiceassistant.py:51]: maybe we should use transformer int4 for the llm for chat as well? @jason-dai

File: python/llm/example/langchain/native_int4/voiceassistant.py [hunk 2]
```diff
@@ -30,6 +30,7 @@
 import speech_recognition as sr
 import pyttsx3
```
Comment (shane-huang) [inline python/llm/example/langchain/native_int4/voiceassistant.py:31]: I think we can make it transformer int4 example. Create a whisper folder and put it there. Add README if necessary. Just follow how it is done for other transformer int4 examples.

Comment (jason-dai) [reply python/llm/example/langchain/native_int4/voiceassistant.py:31]: yes, we can add a whisper example

File: python/llm/example/langchain/native_int4/voiceassistant.py (hunk 1 above)
Comment (jason-dai) [reply python/llm/example/langchain/native_int4/voiceassistant.py:51]: yes, we want to do that if the performance is close

**Comment by leonardozcm** [comment]
@shane-huang whisper examples here

**Comment by shane-huang** [comment]
merge for now.

**Review by shane-huang — APPROVED** [review]

**Pull request merged**